### PR TITLE
FWI-1261 - Remove OPA CRD

### PR DESCRIPTION
**Pull request by makoscafee**
This PR fixes issue [FWI-1261](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-1261)

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### What's the goal of this PR?
Remove CRD dependency for `OPA` and instead use API to get all the checks
### What alternative solution should we consider, if any?

[Internal Ticket FWI-1261](https://fairwinds.myjetbrains.com/youtrack/issue/FWI-1261)

### CODE REVIEW

**Comment by rbren** [comment]
@makoscafee is this still WIP?

**Comment by makoscafee** [comment]
> @makoscafee is this still WIP?

No, it is ready for review

**Review by ivanfetch-fw — APPROVED** [review]
This code looks more clean, and looks good to me. I also tested locally and confirmed I see action-items generated from the OPA plugin.

Given the potential impact, I think it might be nice to have a second set of eyes on this PR CC @rbren.

File: plugins/opa/pkg/opa/opa.go [hunk 1]
```diff
@@ -182,113 +155,6 @@ func getInsightsChecks() (clusterCheckModel, error) {
...
-	logrus.Infof("Updating checks in the cluster")
-	for _, supposedCheck := range jsonResponse.Checks {
-		found := false
-		for _, check := range checks.Items {
-			if check.GetName() == supposedCheck.Name {
-				found = true
-				break
-			}
-		}
-		newCheck := supposedCheck.GetUnstructuredObject(thisNamespace)
```
Comment (ivanfetch-fw) [inline plugins/opa/pkg/opa/opa.go:252]: @makoscafee I believe this method `GetUnstructuredObject` is no longer used and could be deleted? It's in `pkg/opa/types.go`.

**Review by rbren — APPROVED** [review]

File: plugins/opa/pkg/opa/opa.go (hunk 1 above)
Comment (makoscafee) [reply plugins/opa/pkg/opa/opa.go:252]: done

File: plugins/opa/pkg/opa/types.go [hunk 2]
```diff
@@ -69,12 +68,6 @@ func (o *OutputFormat) SetDefaults(others ...OutputFormat) {
...
 }
 
-// CustomCheck is a custom OPA check.
-type CustomCheck struct {
-	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
-	Spec              CustomCheckSpec
-}
-
 // CustomCheckSpec is the body of a Custom Check object
 type CustomCheckSpec struct {
```
Comment (rbren) [inline plugins/opa/pkg/opa/types.go:72]: I think we can remove this one too now, right?

Comment (makoscafee) [reply plugins/opa/pkg/opa/types.go]: correct. Done

**Review by rbren — APPROVED** [review]

**Pull request merged**